renderer/src/features/clients/hooks/use-manage-clients.ts

```diff
@@ -0,0 +1,210 @@
+    []
+
+  const { mutateAsync: registerClient } = useToastMutation({
+    ...postApiV1BetaClientsMutation(),
+    onSettled: () => {
+      queryClient.invalidateQueries({
+        queryKey: getApiV1BetaDiscoveryClientsQueryKey(),
+      })
+      queryClient.invalidateQueries({
+        queryKey: ['api', 'v1beta', 'groups'],
+      })
+      queryClient.invalidateQueries({
+        queryKey: getApiV1BetaClientsQueryKey(),
+      })
+    },
+  })
+
+  const { mutateAsync: unregisterClient } = useToastMutation({
+    mutationFn: async ({
+      clientType,
+      groupName,
+    }: {
+      clientType: string
+      groupName: string
+    }) => {
+      return await deleteApiV1BetaClientsByNameGroupsByGroup({
+        path: {
+          name: clientType,
+          group: groupName,
+        },
+        parseAs: 'text',
+        responseStyle: 'data',
+        throwOnError: true,
+      })
+    },
+    onSettled: () => {
+      queryClient.invalidateQueries({
+        queryKey: getApiV1BetaDiscoveryClientsQueryKey(),
+      })
+      queryClient.invalidateQueries({
+        queryKey: ['api', 'v1beta', 'groups'],
+      })
+    },
+  })
+
+  const addClientToGroup = async (clientType: string, groupName: string) => {
+    const client = installedClients.find((c) => c.client_type === clientType)
+    if (!client) {
+      throw new Error(`Client ${clientType} is not installed or available`)
+    }
+
+    const currentClients = await getApiV1BetaClients({
+      parseAs: 'text',
+      responseStyle: 'data',
+    })
+
+    const parsed =
+      typeof currentClients === 'string'
+        ? currentClients
+          ? JSON.parse(currentClients)
+          : null
+        : currentClients
+
+    const list: Array<{ name?: { name?: string }; groups?: string[] }> =
+      Array.isArray(parsed)
+        ? parsed
+        : parsed && Array.isArray((parsed as { clients?: unknown[] }).clients)
+          ? (((parsed as { clients?: unknown[] }).clients || []) as Array<{
+              name?: { name?: string }
+              groups?: string[]
+            }>)
+          : []
+
+    const currentClient = list.find(
+      (clientItem) => clientItem.name?.name === clientType
+    )
+    const existingGroups = currentClient?.groups || []
+
+    const newGroups = existingGroups.includes(groupName)
+      ? existingGroups
+      : [...existingGroups, groupName]
+
+    await registerClient({
+      body: {
+        name: clientType,
+        groups: newGroups,
+      },
+    })
+
+    trackEvent(`Client ${clientType} registered`, {
+      client: clientType,
+      groups: newGroups,
+    })
+  }
+
+  const removeClientFromGroup = async (
+    clientType: string,
+    groupName: string
+  ) => {
+    // Find the client in the installed clients list
+    const client = installedClients.find((c) => c.client_type === clientType)
+    if (!client) {
+      throw new Error(`Client ${clientType} is not installed or available`)
```

Comment:
is this error catched?

Reply:
(again, i had to simluate this error because normally it would never happen)